kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/LlvmModuleSpecificationImpl.kt

```diff
@@ -28,10 +29,18 @@ internal abstract class LlvmModuleSpecificationBase(protected val cachedLibrarie
     override fun containsPackageFragment(packageFragment: IrPackageFragment): Boolean =
             packageFragment.konanLibrary.let { it == null || containsLibrary(it) }
 
-    override fun containsDeclaration(declaration: IrDeclaration): Boolean =
-            declaration.konanLibrary.let { it == null || containsLibrary(it) }
-                    // When producing per-file caches, some declarations might be generated by the stub generator.
-                    && declaration.getPackageFragment() !is IrExternalPackageFragment
+    private val containsCache = mutableMapOf<IrDeclaration, Boolean>()
+
+    override fun containsDeclaration(declaration: IrDeclaration): Boolean = containsCache.getOrPut(declaration) {
+        val metadata = ((declaration as? IrMetadataSourceOwner)?.metadata as? KonanMetadata)
```

Comment:
This drops the check that the declarations package fragment is not an `IrExternalPackageFragment` and the comment explaining why that is needed. Is that on purpose?

Reply:
Ah missed that, I'll add that back in.